WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingUseCase.swift

```diff
@@ -101,6 +101,16 @@ final class CardPresentPaymentsOnboardingUseCase: CardPresentPaymentsOnboardingU
             savePreferredPlugin(selectedPlugin)
         }
     }
+
+    func clearPluginSelection() {
+        guard let siteID = siteID else {
+            return
+        }
+        preferredPluginLocal = nil
+        let action = AppSettingsAction.forgetPreferredInPersonPaymentGateway(siteID: siteID)
```

Comment:
Not a blocker but, don't we need to wait for completion here to `updateState()`? 😃 Is it done synchronously?

I see that here in `AppSettingsStore`

```
func forgetPreferredInPersonPaymentGateway(siteID: Int64) {
        let storeSettings = getStoreSettings(for: siteID)
        let newSettings = storeSettings.copy(preferredInPersonPaymentGateway: .some(nil))
        setStoreSettings(settings: newSettings, for: siteID, onCompletion: nil)
    }
```

We have that completion callback 🤔

Reply:
It is synchronous. I agree that it would be more correct to have a completion block, since there's no guarantee that the saving will always be synchronous,  but I'm also hoping to replace this at some point with the direct access introduced in #7012